service/controller/clusterapi/v29/adapter/guest_auto_scaling_group.go

```diff
@@ -64,14 +64,9 @@ func (a *GuestAutoScalingGroupAdapter) Adapt(cfg Config) error {
 	a.HealthCheckGracePeriod = gracePeriodSeconds
 	a.RollingUpdatePauseTime = rollingUpdatePauseTime
 
-	azs, err := key.StatusAvailabilityZones(cfg.MachineDeployment)
-	if err != nil {
-		return microerror.Mask(err)
-	}
-
-	for i, az := range azs {
-		a.PrivateSubnets = append(a.PrivateSubnets, key.PrivateSubnetName(i))
-		a.WorkerAZs = append(a.WorkerAZs, az.Name)
+	for _, az := range cfg.TenantClusterAvailabilityZones {
+		a.PrivateSubnets = append(a.PrivateSubnets, key.PrivateSubnetName(az))
+		a.WorkerAZs = append(a.WorkerAZs, az)
```

Comment:
I'm thinking if we maybe want to remove distinction between Master and Worker AZs and move towards Cluster AZs?